sdk/include/game-contract-sdk/game_base.hpp

```diff
@@ -169,20 +169,18 @@ class game: public eosio::contract {
 
 protected:
     /* utility helpers */
-    uint128_t rand_u128(const checksum256& rand) const {
-        const auto& arr = rand.get_array();
-        // use % operation to save original distribution
-        const uint128_t left = arr[0] % UINT64_MAX;
-        const uint128_t right = arr[1] % UINT64_MAX;
-
-        // just concat parts
-        // it's not fair way(don't save original distribution), but more simpler
-        return (left << 64) & right;
-    }
+    template<typename T>
+    static T cut_to(const checksum256& input) {
```

Comment:
it doesn't save original distribution, prev realization save original distribution for u128 -> u64, and provide simple way to convert u256 -> u128 (not ideal but simple and predictable about distribution)

Reply:
Mb keep old algorithm using recursive templates, with special realization for u256->u128 and templated conversation for another cases.

Reply:
and also we need to accept only unsigned types(filter via type traits `typename = std::enable_if_t<std::is_unsigned<Type>::value>`)